### ISSUE TITLE
Method 'julianDateToGregorianDate' does not work well if the Julian date is before March 1 in a year that is leap only in the Julian calendar

### ISSUE THREAD

**Issue by adrianbranescu** [Adrian-s-Java-Programs/javafx-easter-date-calculator#2]
There are years which are leap years only in the Julian calendar, but not in the Gregorian calendar.

These years are the ones that divide by 100 without remainder but do not divide by 400 without remainder (for example 1700, 1800, 1900 and 2100, but not 2000). 

Such Julian years have one extra day (which is February 29), and the conversion algorithm needs to add a correction in this case.
The problem occurs only for these years and only for dates in January and February.

Since the current application converts only Eastern Easter dates from Julian to Gregorian calendar and since the earliest Eastern Easter Julian date is March 22, the conversion issue does not manifest itself and the conversion of Eastern Easter date from Julian date to Gregorian date always works correctly. For this reason, I am not labeling this issue as a bug, because it has no functional impact on the application.

However, the conversion method gives the impression that it can convert any Julian date, which is not always the case, so this issue should be addressed.

**Comment by adrianbranescu** [comment]
This is a conversion formula found on Wikipedia:
![ConversionFormula](https://user-images.githubusercontent.com/59761765/97767106-a3731b80-1b22-11eb-8739-6550d851ab1c.png)

**Comment by adrianbranescu** [comment]
I have added code that allows a correct conversion from a Julian date to a Gregorian date.

This application was working fine without it, but I thought someone might need a full Julian to Gregorian date converter and I decided to provide the necessary code. The conversion method works with both BC and AD dates.

Since there are years which are leap in Julian calendar, but not in Gregorian calendar, we cannot use LocalDate type to hold a Julian date, because LocalDate is for Gregorian dates and will reject values such as 29-February-1800 or 29-February-1900, which are valid Julian dates, but not valid Gregorian dates.

As a result, I created a separate class (called JulianDate) that can hold a Julian date, and in that class I wrote the code used for conversion to Gregorian date.

For Easter calculation, I have modified the code to work using this new JulianDate type.
